gremlin-go/driver/traversal.go

```diff
@@ -19,41 +19,54 @@ under the License.
 
 package gremlingo
 
+type Traverser struct {
+	bulk  int64
+	value interface{}
+}
+
 type Traversal struct {
 	graph               *Graph
 	traversalStrategies *TraversalStrategies
 	bytecode            *bytecode
+	traverser           *Traverser
+	remote              *DriverRemoteConnection
 }
 
-const host string = "localhost"
-const port int = 8182
-
+// ToList returns the result in a list
 // TODO use TraversalStrategies instead of direct remote after they are implemented
-var remote = DriverRemoteConnection{
-	client: NewClient(host, port),
+func (t *Traversal) ToList() ([]*Result, error) {
+	results, err := t.remote.SubmitBytecode(t.bytecode)
+	if err != nil {
+		return nil, err
+	}
+	return results.All(), nil
 }
 
-// ToList returns the result in a list
-// TODO commenting out as remote.Submit doesn't support bytecode yet
-//func (t *Traversal) ToList() []*Result {
-//	results, _ := remote.Submit(t.bytecode)
-//	return results.All()
-//}
-
 // ToSet returns the results in a set.
 // TODO Go doesn't have sets, determine the best structure for this
-//func (t *Traversal) ToSet() map[*Result]bool {
-//	set := make(map[*Result]bool)
-//	results, _ := remote.Submit(t.bytecode)
-//	for _, r := range results.All() {
-//		set[r] = true
-//	}
-//	return set
-//}
-
-// Iterate iterate all the Traverser instances in the traversal and returns the empty traversal
-// TODO complete this when Traverser is implemented
-func (t *Traversal) Iterate() *Traversal {
-	t.bytecode.addStep("none")
-	return t
+func (t *Traversal) ToSet() (map[*Result]bool, error) {
+	set := map[*Result]bool{}
+	results, err := t.remote.SubmitBytecode(t.bytecode)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range results.All() {
+		set[r] = true
+	}
+	return set, nil
+}
+
+// Iterate all the Traverser instances in the traversal and returns the empty traversal
+func (t *Traversal) Iterate() (*Traversal, error) {
```

Comment:
Do we also need to implement Next?

Reply:
I will add a ticket to do this later since the python implementation is a little screwy and the result sets we are getting are also a littler screwy.